### PR TITLE
ARROW-16264: [C++][CI] Valgrind timeout in arrow-compute-hash-join-node-test

### PR DESCRIPTION
**Pull request by westonpace**

### CODE REVIEW

**Comment by westonpace** [comment]
@github-actions crossbow submit test-conda-cpp-valgrind

File: cpp/src/arrow/compute/exec/bloom_filter_test.cc [hunk 1]
```diff
@@ -364,8 +364,13 @@ void TestBloomLarge(BloomFilterBuildStrategy strategy, int64_t num_build, bool u
 
 TEST(BloomFilter, Basic) {
   std::vector<int64_t> num_build;
+#if defined(THREAD_SANITIZER) or defined(ARROW_VALGRIND)
```
Comment (kou) [inline cpp/src/arrow/compute/exec/bloom_filter_test.cc:367]: Could you use `||` instead of `or` in `#if`?

**Comment by westonpace** [comment]
@github-actions crossbow submit test-conda-cpp-valgrind

File: cpp/src/arrow/compute/exec/bloom_filter_test.cc (hunk 1 above)
Comment (westonpace) [reply cpp/src/arrow/compute/exec/bloom_filter_test.cc]: Yes.  I made this change.  Thank you.

**Comment by westonpace** [comment]
I rebased this on top of the tpc-h generator cleanup since those valgrind failures were blocking the test of this feature.  I converted this to draft as we should merge the other fix first and then I'll convert it back to ready-for-review.

**Comment by westonpace** [comment]
CC @michalursa Valgrind started timing out on the hash-join-node-test.  The root cause for the newly emerged failure may have been the addition of the bloom filter tests or it may have been that I accidentally made the hash join node tests slow.  Either way, I think it's probably fine to shorten the size of inputs we use when testing via valgrind.

**Comment by westonpace** [comment]
@github-actions crossbow submit test-conda-cpp-valgrind

**Comment by westonpace** [comment]
I'm going to move forward with this to unblock valgrind.  If we decide later we want to extend the test set on valgrind we can fix that in a follow-up.

**Pull request closed without merging**